arch/risc-v/src/common/riscv_exception_common.S

```diff
@@ -49,7 +49,9 @@ exception_common:
   addi       sp, sp, -XCPTCONTEXT_SIZE
 
   REGSTORE   x1,  REG_X1(sp)   /* ra */
+#ifdef RISCV_SAVE_GP
```

Comment:
where define this macro?

Reply:
No-where. But it is used elsewhere, like in riscv_vfork.c . And when the gp is not initialized there when creating a new task, it is also wrong to restore gp in here. I used the same macro in here so it can be enabled in the future